### PR TITLE
Fix small bugs in Synthetic Turbulence

### PR DESCRIPTION
**Pull request by gantech**
1. Fix bug in conversion of wind direction to transformation matrix for local coordinate system
2. Fix bug in interpolation of wind perturbation from netcdf file

### CODE REVIEW

File: amr-wind/physics/SyntheticTurbulence.cpp [hunk 1]
```diff
@@ -449,15 +448,16 @@ SyntheticTurbulence::SyntheticTurbulence(const CFDSim& sim)
     //
     // x-direction points to flow direction (convert from compass direction to
     // vector)
-    m_turb_grid.tr_mat.xx() = -std::sin(wind_direction);
-    m_turb_grid.tr_mat.xy() = -std::cos(wind_direction);
+    m_turb_grid.tr_mat.xx() = std::cos(theta_t);
```
Comment (sayerhs) [inline amr-wind/physics/SyntheticTurbulence.cpp:451]: What's the bug here?

Comment (gantech) [reply amr-wind/physics/SyntheticTurbulence.cpp:451]: For some reason, the second row of `tr_mat` came out to zero. This seemed to fix that problem.

Comment (sayerhs) [reply amr-wind/physics/SyntheticTurbulence.cpp:451]: Seems like there bug is in there cross product and not here.

File: amr-wind/physics/SyntheticTurbulence.cpp [hunk 2]
```diff
@@ -449,15 +448,16 @@ SyntheticTurbulence::SyntheticTurbulence(const CFDSim& sim)
...
     m_turb_grid.tr_mat.xz() = 0.0;
+    m_turb_grid.tr_mat.yx() = -std::sin(theta_t);
+    m_turb_grid.tr_mat.yy() = std::cos(theta_t);
+    m_turb_grid.tr_mat.yz() = 0.0;
     // z always points upwards (for now...)
     m_turb_grid.tr_mat.zx() = 0.0;
     m_turb_grid.tr_mat.zy() = 0.0;
     m_turb_grid.tr_mat.zz() = 1.0;
-    // y = z .cross. x
-    m_turb_grid.tr_mat.y() = m_turb_grid.tr_mat.z() ^ m_turb_grid.tr_mat.x();
```
Comment (sayerhs) [inline amr-wind/physics/SyntheticTurbulence.cpp:460]: @gantech See #400 for a fix to catch this error in future.

File: amr-wind/physics/SyntheticTurbulence.cpp (hunk 1 above)
Comment (sayerhs) [reply amr-wind/physics/SyntheticTurbulence.cpp:451]: Also you can simplify this section by writing 

```C++
const amrex::Real wdir = 270.0 - wind_direction; 
m_turb_grid.tr_mat = Tensor::zrot(wdir);
```

File: amr-wind/physics/SyntheticTurbulence.cpp (hunk 2 above)
Comment (gantech) [reply amr-wind/physics/SyntheticTurbulence.cpp:460]: Thanks Shreyas.

File: amr-wind/physics/SyntheticTurbulence.cpp (hunk 1 above)
Comment (gantech) [reply amr-wind/physics/SyntheticTurbulence.cpp]: Thanks Shreyas. Fixed in https://github.com/Exawind/amr-wind/pull/399/commits/57de3963a2bf9632042441ceed12f74cde912cf5

**Review by sayerhs — APPROVED** [review]

**Pull request merged**